Dockerfile

```diff
@@ -1,33 +1,249 @@
+    done && \
+    for static_l in ${FSLDIR}/extras/include/boost/bin.v2/libs/*/build/gcc-4.8.5/release/link-static/threading-multi/*.[ao] ${FSLDIR}/extras/include/boost/bin.v2/libs/log/build/gcc-4.8.5/release/link-static/log-api-unix/threading-multi/libboost_log*.a; do \
+      l=${FSLDIR}/extras/lib/$(basename ${static_l}); \
+      [ -f $l ] && diff ${l} ${static_l} \
+        && rm -r ${l} \
+        && echo "ln -s ${static_l} ${l}" >> /create_links.sh ; \
+    done && \
+    for r in ${FSLDIR}/data/xtract_data/standard/F99/surf/rh*; do \
+      diff ${r} ${r/rh./R.} \
+        && rm -r ${r} \
+        && echo "ln -s ${r/rh./R.} ${r}" >> /create_links.sh ; \
+    done && \
+    for l in ${FSLDIR}/data/xtract_data/standard/F99/surf/lh*; do \
+      diff ${l} ${l/lh./L.} \
+        && rm -r ${l} \
+        && echo "ln -s ${l/lh./L.} ${l}" >> /create_links.sh ; \
+    done && \
+    diff ${FSL_PYTHON}/share/jupyter/nbextensions/jupyter-js-widgets \
+        ${FSL_PYTHON}/lib/python3.7/site-packages/widgetsnbextension/static \
+      && rm -r ${FSL_PYTHON}/share/jupyter/nbextensions/jupyter-js-widgets \
+      && echo "ln -s ${FSL_PYTHON}/lib/python3.7/site-packages/widgetsnbextension/static ${FSL_PYTHON}/share/jupyter/nbextensions/jupyter-js-widgets" >> /create_links.sh && \
+    #diff ${FSL_PYTHON}/lib/python3.7/ensurepip/_bundled ${FSLDIR}/fslpython/lib/python3.9/ensurepip/_bundled \
+      rm -r ${FSLDIR}/fslpython/lib/python3.9/ensurepip/_bundled \
+      && echo "ln -s ${FSL_PYTHON}/lib/python3.7/ensurepip/_bundled ${FSLDIR}/fslpython/lib/python3.9/ensurepip/_bundled" >> /create_links.sh && \
+    rm -r ${FSLDIR}/lib/libbedpostx_cuda.so && \
+    rm -r ${FSLDIR}/lib/libvtk* ${FSLDIR}/lib/libqwt.* ${FSLDIR}/lib/libfslvtkio.*\
+  '     # end of "bash -0 extglob..."
+#############
+###  Now, get a new machine with only the essentials  ###
+FROM python:${BASE_PYTHON_VERSION}-slim-${DEBIAN_VERSION} as Application
+# This makes the BASE_PYTHON_VERSION available inside this stage
+ARG BASE_PYTHON_VERSION
+ENV PYTHON_LIB_PATH=/usr/local/lib/python${BASE_PYTHON_VERSION}
+ENV FSLDIR=/usr/local/fsl/ \
+    FSLOUTPUTTYPE=NIFTI_GZ
+ENV PATH=${FSLDIR}/bin:$PATH \
+    LD_LIBRARY_PATH=${FSLDIR}:${LD_LIBRARY_PATH}
+# Copy system binaries and libraries:
+COPY --from=builder ./lib/x86_64-linux-gnu/     /lib/x86_64-linux-gnu/
+COPY --from=builder ./usr/lib/x86_64-linux-gnu/ /usr/lib/x86_64-linux-gnu/
+COPY --from=builder ./usr/bin/                  /usr/bin/
+COPY --from=builder ./usr/local/bin/           /usr/local/bin/
+# COPY --from=builder ./${PYTHON_LIB_PATH}/site-packages/      ${PYTHON_LIB_PATH}/site-packages/
+# Copy $FSLDIR:
+COPY --from=builder ./${FSLDIR}/  ${FSLDIR}/
+# Create the links:
+COPY --from=builder ./create_links.sh  /create_links.sh
+RUN chmod u+x /create_links.sh && \
+    /create_links.sh
+
+#RUN curl https://packages.bic.mni.mcgill.ca/mni-models/icbm152/mni_icbm152_nl_VI_nifti.zip --output /tmp/mni.zip && \
```

Comment:
Do we want to dispose of these lines, since we aren't incorporating the template?

Reply:
I want to leave these in so that it's easier to see what was changed from the original if more changes are needed in the future.